test/functional/outputs/azuremonitor/forward_to_azuremonitor_test.go

```diff
@@ -0,0 +1,194 @@
+				constants.SharedKey: sharedKey,
+			},
+		)
+		framework.Secrets = append(framework.Secrets, secret)
+		Expect(framework.DeployWithVisitor(func(b *runtime.PodBuilder) error {
+			return mockServer.NewVisitor(b, framework)
+		})).To(BeNil())
+	})
+
+	AfterEach(func() {
+		framework.Cleanup()
+	})
+
+	It("should accept application logs", func() {
+		// Write application logs
+		timestamp := "2020-11-04T18:13:59.061892+00:00"
+		message := "This is my test message"
+		applicationLogLine := functional.NewCRIOLogMessage(timestamp, message, false)
+		Expect(framework.WriteMessagesToApplicationLog(applicationLogLine, 1)).To(BeNil())
+		time.Sleep(30 * time.Second)
+
+		//read log from collector container
+		collectorLog, err := framework.ReadCollectorLogs()
+		Expect(err).To(BeNil())
+		Expect(strings.Count(collectorLog, failedReason)).To(BeEquivalentTo(0))
+
+		//read log from mock server container and validate it
+		mockoonLogs := readLogFromMockoon(framework)
+		Expect(mockoonLogs).ToNot(BeNil())
+		Expect(len(mockoonLogs)).To(BeEquivalentTo(2))
+		Expect(mockoonLogs[0].Message).To(BeEquivalentTo(serverStarted))
+		Expect(mockoonLogs[1].RequestMethod).To(BeEquivalentTo(http.MethodPost))
+		Expect(mockoonLogs[1].Message).To(BeEquivalentTo(transactionRecorded))
+		Expect(mockoonLogs[1].Transaction.Request.Body).ToNot(BeEmpty())
+		var appLogs []types.ApplicationLog
+		err = types.ParseLogsFrom(utils.ToJsonLogs([]string{mockoonLogs[1].Transaction.Request.Body}), &appLogs, false)
+		Expect(err).To(BeNil())
+		Expect(len(appLogs)).To(BeEquivalentTo(1))
+		validateAppLog(appLogs[0], message, framework)
+	})
+
+	It("should accept batch application logs", func() {
+		// Write application logs
+		timestamp := "2020-11-04T18:13:59.061892+00:00"
+		message := "This is my new test message"
+		applicationLogLine := functional.NewCRIOLogMessage(timestamp, message, false)
+		Expect(framework.WriteMessagesToApplicationLog(applicationLogLine, 3)).To(BeNil())
+		time.Sleep(30 * time.Second)
+
+		//read log from collector container
+		collectorLog, err := framework.ReadCollectorLogs()
+		Expect(err).To(BeNil())
+		Expect(strings.Count(collectorLog, failedReason)).To(BeEquivalentTo(0))
+
+		//read log from mock server container and validate it
+		mockoonLogs := readLogFromMockoon(framework)
+		Expect(mockoonLogs).ToNot(BeNil())
+		Expect(len(mockoonLogs)).To(BeEquivalentTo(2))
+		Expect(mockoonLogs[0].Message).To(BeEquivalentTo(serverStarted))
+		Expect(mockoonLogs[1].RequestMethod).To(BeEquivalentTo(http.MethodPost))
+		Expect(mockoonLogs[1].Message).To(BeEquivalentTo(transactionRecorded))
+		Expect(mockoonLogs[1].Transaction.Request.Body).ToNot(BeEmpty())
+		var appLogs []types.ApplicationLog
+		err = types.ParseLogsFrom(utils.ToJsonLogs([]string{mockoonLogs[1].Transaction.Request.Body}), &appLogs, false)
+		Expect(err).To(BeNil())
+		Expect(len(appLogs)).To(BeEquivalentTo(3))
+		for i := 0; i < 3; i++ {
+			validateAppLog(appLogs[i], message, framework)
```

Comment:
Consider using templates like [these](https://github.com/openshift/cluster-logging-operator/blob/release-5.8/test/functional/normalization/message_format_test.go#L41) to validate the entire message format

Reply:
done